pkg/wasm/httpfetcher.go

```diff
@@ -15,16 +15,23 @@
 package wasm
 
 import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
 	"context"
 	"crypto/tls"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"net/http"
 	"time"
 
 	"github.com/cenkalti/backoff/v4"
 )
 
+var tarMagicNumber = []byte{0x75, 0x73, 0x74, 0x61, 0x72}
```

Comment:
just for my learning, from where you got these magic numbers?

Reply:
You can check them various documents. (e.g., https://en.wikipedia.org/wiki/List_of_file_signatures)